### PR TITLE
Fix -fpermissive issues, remove workaround flags

### LINKED ISSUE #47: Remove -fpermissive
#20 introduced "-fpermissive", a flag that turns non-conformant code errors into warnings, in order to be able to build on Linux.

We should manually correct these errors and disable the flag again.

### PR DESCRIPTION
**Pull request by th1000s**
* Don't take the address of an rvalue
* Remove -fpermissive and -Wno-address-of-temporary flags
 
---

Closes #47

### CODE REVIEW

File: Descent3/ConfigItem.cpp [hunk 1]
```diff
@@ -164,12 +164,14 @@ bool ConfigItem::Create(NewUIGameWindow *parentwnd, int type, int flags, int x,
     Int3();
   }
 
-  if (type == CIT_SLIDER) // slider's need to have the label centered vertically
-    m_tLabel.Create(parentwnd, &UITextItem(label, UICOL_TEXT_NORMAL), x, y + 7, UIF_FIT);
-  else {
+  if (type == CIT_SLIDER) { // slider's need to have the label centered vertically
+    auto item = UITextItem(label, UICOL_TEXT_NORMAL);
```
Comment (Lgt2x) [inline Descent3/ConfigItem.cpp:168]: Someone was recommanding `UITextItem item{label, UICOL_TEXT_NORMAL}` which is even shorter and avoids using `auto` https://github.com/kevinbentley/Descent3/issues/47#issuecomment-2064817034

File: Descent3/ctlconfig.cpp [hunk 2]
```diff
@@ -882,14 +886,14 @@ void wpnsel_cfg_screen::realize() {
...
-  m_primary_low.Create(m_menu, &UITextItem(MONITOR9_NEWUI_FONT, TXT_LOWPRIORITY), WPNSEL_PRIMARY_X,
-                       WPNSEL_SECONDARY_Y - 11);
-  m_secondary_low.Create(m_menu, &UITextItem(MONITOR9_NEWUI_FONT, TXT_LOWPRIORITY), WPNSEL_SECONDARY_X,
-                         WPNSEL_SECONDARY_Y - 11);
+  auto item = UITextItem(MONITOR9_NEWUI_FONT, TXT_PRIMARY);
+  m_primary_title.Create(m_menu, &item, WPNSEL_PRIMARY_X, WPNSEL_PRIMARY_Y - 28);
+  auto item2 = UITextItem(MONITOR9_NEWUI_FONT, TXT_SECONDARY);
+  m_secondary_title.Create(m_menu, &item, WPNSEL_SECONDARY_X, WPNSEL_SECONDARY_Y - 28);
+  auto item3 = UITextItem(MONITOR9_NEWUI_FONT, TXT_LOWPRIORITY);
+  m_primary_low.Create(m_menu, &item, WPNSEL_PRIMARY_X, WPNSEL_SECONDARY_Y - 11);
```
Comment (Lgt2x) [inline Descent3/ctlconfig.cpp:894]: the reference should &item3

File: Descent3/ctlconfig.cpp [hunk 3]
```diff
@@ -882,14 +886,14 @@ void wpnsel_cfg_screen::realize() {
...
-                         WPNSEL_PRIMARY_Y - 28);
-  m_secondary_title.Create(m_menu, &UITextItem(MONITOR9_NEWUI_FONT, TXT_SECONDARY), WPNSEL_SECONDARY_X,
-                           WPNSEL_SECONDARY_Y - 28);
-  m_primary_low.Create(m_menu, &UITextItem(MONITOR9_NEWUI_FONT, TXT_LOWPRIORITY), WPNSEL_PRIMARY_X,
-                       WPNSEL_SECONDARY_Y - 11);
-  m_secondary_low.Create(m_menu, &UITextItem(MONITOR9_NEWUI_FONT, TXT_LOWPRIORITY), WPNSEL_SECONDARY_X,
-                         WPNSEL_SECONDARY_Y - 11);
+  auto item = UITextItem(MONITOR9_NEWUI_FONT, TXT_PRIMARY);
+  m_primary_title.Create(m_menu, &item, WPNSEL_PRIMARY_X, WPNSEL_PRIMARY_Y - 28);
+  auto item2 = UITextItem(MONITOR9_NEWUI_FONT, TXT_SECONDARY);
```
Comment (Lgt2x) [inline Descent3/ctlconfig.cpp:891]: could be named itemPrimaryTitle to make sure the function below is using the right one (which is not the case). The same goes in other places where multiple items are used

File: Descent3/newui.cpp [hunk 4]
```diff
@@ -586,8 +586,8 @@ void DoWaitPopup(bool enable, char *message) {
         grtext_SetFont(MONITOR9_NEWUI_FONT);
         w = grtext_GetTextLineWidth(msg);
         h = grtext_GetTextHeight(msg);
-        text.Create(msgbox, &UITextItem(MONITOR9_NEWUI_FONT, msg, NEWUI_MONITORFONT_COLOR), (msgbox->W() - w) / 2,
-                    (msgbox->H() - h) / 2);
+        auto item = UITextItem(MONITOR9_NEWUI_FONT, msg, NEWUI_MONITORFONT_COLOR);
+        text.Create(msgbox, &item, (msgbox->W() - w) / 2, (msgbox->H() - h) / 2);
```
Comment (Lgt2x) [inline Descent3/newui.cpp:590]: &item4

File: Descent3/ctlconfig.cpp (hunk 2 above)
Comment (th1000s) [reply Descent3/ctlconfig.cpp]: Oh, thanks for catching that!

**Review by Lgt2x — COMMENTED** [review]
One more reference to change and we're good to merge, thanks for taking care of this !

**Comment by th1000s** [comment]
Thanks for taking a _good_ look at it! :smile:

**Pull request merged**